cmake/DetectOptions.cmake

```diff
@@ -179,11 +207,15 @@ if(ADIOS2_USE_CUDA)
   else()
     find_package(CUDAToolkit REQUIRED)
   endif()
+  if(CMAKE_CUDA_COMPILER AND CUDAToolkit_FOUND)
+    enable_language(CUDA)
+    set(ADIOS2_HAVE_CUDA TRUE)
+    set(ADIOS2_HAVE_GPU_Support TRUE)
+  endif()
 endif()
-if(CMAKE_CUDA_COMPILER AND CUDAToolkit_FOUND)
-  enable_language(CUDA)
-  set(ADIOS2_HAVE_CUDA TRUE)
-  set(ADIOS2_HAVE_GPU_Support TRUE)
+
+if(ADIOS_HAVE_Kokkos AND ADIOS_HAVE_CUDA)
+    message(FATAL_ERROR "The Kokkos and CUDA backends cannot be active concurrently")
```

Comment:
```suggestion
```

Reply:
Having this would prevent mistakes if we change the way we find different backends (like the one in PR #3485)

Reply:
The problem is that when that time comes we will have to debug this, why not just leave it?

Reply:
In the same line, what if the future code changes and we now support both Kokkos and CUDA, will have to remove this as well.

Reply:
We will never support both

Reply:
I still do not see the reason of writing conditional for potential future code, but you might have some changes in mind here for the future and this is a noop, Ill resolve this.